doc/user_manual/code_interfaces/generic.tex

```diff
@@ -156,3 +156,90 @@ \subsection{Generic Interface}
+                                                            %
+                                                            These variables are inputted in a single
+                                                            \xmlNode{variables} XML node:
+                                                            \begin{itemize}
+                                                              \item \xmlNode{variables}, \xmlDesc{comma separated list, required parameter}, in the body of
+                                                              this XML node, the user needs to specify the name of the variables (separated by commas).
+                                                              %
+                                                              This variables need to match variables used/defined in the external python
+                                                              function.
+                                                              %
+                                                            \end{itemize}
+                                                            When the external function variables are defined, at runtime, RAVEN initializes
+                                                            them and keeps track of their values during the simulation.
+                                                            %
+                                                            Each variable defined in the \xmlNode{stoppingCriteriaFunction} block is available in the
+                                                            function as a python \texttt{raven.} member. In the following, an example of a
+                                                            user-defined stopping condition function is reported.
+\begin{lstlisting}[language=python]
+def criteria(raven):
+  if raven.var1[-1] < 1.0 or raven.var2[-1] < 1.0:
+    return False
+  else:
+    return True
+\end{lstlisting}
+\end{itemize}
+
+Then, our example XML for the code would be
+
+Example:
+\begin{lstlisting}[style=XML]
+    <Code name="poly" subType="GenericCode">
+      <executable>GenericInterface/poly_inp.py</executable>
+      <inputExtentions>.one,.two</inputExtentions>
+      <clargs   type='prepend' arg='python'/>
+      <clargs   type='input'   arg='-i'  extension='.one'/>
+      <fileargs type='input'   arg='two' extension='.two'/>
+      <outputFile>fixed_output.csv</outputFile>
+      <!--
+        Check the output every 15 seconds
+       -->
+      <onlineStopCriteriaTimeInterval>
+        15.0
+      </onlineStopCriteriaTimeInterval>
+      <!-- use the function named criteria in the python file stoppingCriteria.py --> 
+      <stoppingCriteriaFunction file="stoppingCriteria.py" name="criteria">
+         <!-- 
+           variables below must be retrievable
+           from the generic code interface csv output 
+         -->
+        <variables>var1, var2 </variables>
+      </stoppingCriteriaFunction>
```

Comment:
This is a more consistent question. Inside RAVEN, we use <Functions> block to define all functions, and then reference the function name in corresponding entities so that we can use it. In this implementation, the whole function is defined within the entity. I may prefer we keep the consistency with current RAVEN function structure. What do you think? The following is the example: 
<img width="681" alt="image" src="https://github.com/user-attachments/assets/697d8d1d-87ec-4f25-8f96-7de75a7e6af9" />

Reply:
@wangcj05 I addressed your comment using the assembler object system. let me know what you think.